README.md

```diff
@@ -1 +1,238 @@
+To warm up with the async code, consider an example in which we have a few `console.log` statements and a timeout:
+
+```js
+var a = 1
+console.log('The value of a is ', a)
+setTimeout(function(){
+  a = 2
+  console.log('While the value of a is', a)
+}, 500)
+console.log('Now the value of a is', a)
+```
+
+For someone not versed in async way of thinking, it might appear that the resulting output will be
+
+```
+The value of a is  1
+While the value of a is 2
+Now the value of a is 2
+```
+
+Because in most languages the execution goes in the order of the statements, a person might think that because `a =2` precedes `Now the value...`, the second and third output will have the values of 2. This is incorrect, because `setTimeout()` is scheduled asynchronously. The right output will be:
+
+```js
+The value of a is  1
+Now the value of a is 1
+// delay of roughly 500 milliseconds
+While the value of a is 2
+```
+
+This misunderstanding of async code can lead to many bugs. It's important to pay attention to how async works to avoid creating issues for yourself down the line.
+
+
+## fs.readFile
+
+Let's take our synchronous code to read a file, and translate it to asynchronous:
+
+```js
+var fs = require('fs')
+console.log('Start reading file')
+var content = fs.readFileSync('readme.md', 'utf-8')
+console.log('Finish reading file')
+console.log(content)
+```
+
+The code will produce "Start reading file", then wait until the file reading is over and finally print "Finish reading file".
+
+The async method `fs.readFile` is a better because our program can do something else while it waits for the content of the file. There are a few things which are different in `fs.readFile` comparing to `fs.readFileSync`:
+
+1. We need to use callback
+2. The callback accepts an error object as a parameter
+3. The callback accepts the content object (the data from the file) as a parameter
+4. We don't get the content right away because `readFile` is **not an expression** like `fs.readFileSync`. (Expressions are functions which return something.) In other words, we cannot write `var content = fs.readFile(...)`.
+
+```js
+var fs = require('fs')
+console.log('Start reading file')
+fs.readFile('README.md', 'utf-8', function(error, content) {
+  console.log('Finish reading file')
+  console.log(content)
+})
+console.log('I CAN DO SOMETHING ELSE WHILE WAITING!')
+```
+
+You might wonder what would be the output of our async code? 
+
+```
+Start reading file
+I CAN DO SOMETHING ELSE WHILE WAITING!
+Finish reading file
+... // Content of the file
+```
+
+You can observe that while the app was waiting on the file system task (reading), it processed the "I CAN DO SOMETHING ELSE WHILE WAITING!" console log. Typically a Node.js web server will be processing and handling multiple requests from clients at the same time while waiting for an input/output operation to finish.
+
+In the synchronous world, we would handle errors with `try/catch` so that `readFileSync` is wrapped in it:
```

Comment:
This is a rough transition from I/O to error handling -- can we either break this section up a bit or make the transition smoother?